bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/automation/jrubyscripting/internal/JRubyConsoleCommandExtension.java

```diff
@@ -0,0 +1,549 @@
+
+                puts "JRuby #{JRUBY_VERSION}"
+                puts "JRuby Scripting Library #{library_version || 'is not installed'}"
+                puts "GEM_HOME: #{ENV['GEM_HOME']}"
+                puts "RUBYLIB: #{ENV['RUBYLIB']}"
+                    """;
+
+        executeWithFullJRuby(console, engine -> engine.eval(PRINT_VERSION_NUMBERS));
+        console.println("Script path: " + scriptFileWatcher.getWatchPath());
+        console.println("");
+        console.println("JRuby Scripting Add-on Configuration:");
+        console.println("=====================================");
+
+        ConfigDescription configDescription = configDescriptionRegistry
+                .getConfigDescription(URI.create(JRubyScriptEngineFactory.CONFIG_DESCRIPTION_URI));
+
+        if (configDescription == null) {
+            console.println("No configuration found for JRuby Scripting add-on. This is probably a bug.");
+            return;
+        }
+
+        Map<String, String> config = jRubyScriptEngineFactory.getConfiguration().getConfigurations();
+        Map<String, List<ConfigDescriptionParameter>> paramsByGroup = configDescription.getParameters().stream()
+                .collect(Collectors.groupingBy(ConfigDescriptionParameter::getGroupName));
+
+        configDescription.getParameterGroups().forEach(group -> {
+            List<ConfigDescriptionParameter> parameters = paramsByGroup.get(group.getName());
+            if (parameters == null) {
+                return;
+            }
+
+            String groupLabel = group.getLabel();
+            if (groupLabel == null) {
+                groupLabel = group.getName();
+            }
+            console.println(groupLabel);
+            parameters.forEach(parameter -> {
+                console.println("  " + parameter.getName() + ": " + config.get(parameter.getName()));
+            });
+            console.println("");
+        });
+    }
+
+    private Map<String, String> getConsoles() {
+        return (Map<String, String>) executeWithPlainJRuby(null, engine -> engine
+                .eval("require 'openhab/console/registry'; OpenHAB::Console::REGISTRY.transform_keys(&:to_s)"));
+    }
+
+    private void startConsole(Console console, Session session, String[] args) {
+        String script = jRubyScriptEngineFactory.getConfiguration().getConsole();
+        if (args.length > 0) {
+            if ("--list".equals(args[0])) {
+                Map<String, String> consoles = (Map<String, String>) getConsoles();
+                if (consoles == null) {
+                    console.println("No console scripts available. Please install the JRuby helper library gem.");
+                } else {
+                    console.println("Available console scripts:");
+                    String defaultConsole = jRubyScriptEngineFactory.getConfiguration().getConsole();
+                    consoles.forEach((name, description) -> {
+                        if (defaultConsole.endsWith(name)) {
```

Comment:
you should be more discerning here. `if (defaultConsole.equals("openhab/console/" + name)) {`

cause what if I have default console configured to be "ccutrer/irb"? that would match the "irb" from the registry, but is decidedly _not_ the default console.

and along those lines, perhaps if the default console isn't found in the registry, add an extra line at the bottom along the lines of `"  (default) - " + defaultConsole`

Reply:
good catch! refactored.